app/process_proposal.go

```diff
@@ -49,42 +51,40 @@ func (app *App) ProcessProposal(req abci.RequestProcessProposal) (resp abci.Resp
 		ante.DefaultSigVerificationGasConsumer,
 		app.IBCKeeper,
 		app.ParamsKeeper,
-		app.MsgGateKeeper,
+		&app.CircuitKeeper,
+		app.BlockedParamsGovernance(),
 	)
-	sdkCtx := app.NewProposalContext(req.Header)
-	subtreeRootThreshold := appconsts.SubtreeRootThreshold(app.GetBaseApp().AppVersion())
+	blockHeader := ctx.BlockHeader()
+
+	appVersion, err := app.AppVersion(ctx)
+	if err != nil {
+		logInvalidPropBlockError(app.Logger(), blockHeader, "failure to get app version", err)
+		return reject(), nil
+	}
+
+	subtreeRootThreshold := appconsts.SubtreeRootThreshold(appVersion)
```

Comment:
similar question to above, don't we always know we're on v4 if we're running this code? if so, can we just put in the latest instead of looking it up?